src/components/form-elements/draft-js-mention-selector/DraftJSMentionSelector.js

```diff
@@ -87,15 +87,47 @@ class DraftJSMentionSelector extends React.Component<Props, State> {
      * @returns {void}
      */
     componentWillReceiveProps(nextProps: Props) {
```

Comment:
Change this to componentDidMount while you are at it, unless it causes obvious probs.

Reply:
Not sure if componentDidMount will be enough since we need to inspect the `editorState` coming in from `ApprovalCommentForm` each time a change is made to the DraftJS component. Maybe in combination with componentDidUpdate but leaving it as is should save extra renders right?

Reply:
It looks like we'd need a combination of `getDerivedStateFromProps` to handle what `state` should become and `componentDidMount` to determine whether or not to run `this.checkValidity()`, depending on what changed. I think doing so would also eliminate much of the code in `this.handleChange`, as well.